### ISSUE TITLE
[improvement] "Close all to the left/right" and unsaved tabs

### ISSUE THREAD

**Issue by AFract** [notepad-plus-plus/notepad-plus-plus#2555]
Hello, 
Currently when pushing for example the "Close all to the right" button in the tabs, if there is at least one unsaved new file, Notepad++ asks to save yes / no / cancel it. 
If you choose Cancel, the tab stays (as expected) open, but also all the other tabs which should have been closed while using "Close all to the right". 
It's annoying when you want to close at one all "saved" tabs but there is still one or more unsaved ones : for me all the "saved" ones should have been closed, even if pressing Cancel for the unsaved one(s).
Thank you

**Comment by MetaChuh** [comment]
please update your notepad++ to the latest version [<kbd>Notepad++ 7.6.6</kbd>](https://notepad-plus-plus.org/download/v7.6.6.html) and migrate or reinstall your plugins if you update from version 7.5.9 or below.
if your issue still persists, and it is still of relevance to you, please feel free to file a new issue as easy to understand and reproducible as possible.

thank you for your help and best regards.

-----
```
general notes:

please close your issues as soon as you are satisfied, to help
keeping the notepad++ issue tracker clean.

when creating a new issue, please provide all information as shown at the issue template.
your notepad++ debug info is part of the minimum requirements.
(? > Debug Info... > Copy debug info into clipboard).

please also conduct a search at the issue tracker beforehand, to check if a similar issue
already exists.

for general questions, or if it is not sure, whether your issue is directly related to
the notepad++ source code, please visit us at the notepad++ community forum, and
search if related topics exists. you are welcome to post either at similar
topics, or to create a new topic at Help Wanted or General Discussion.
(no extra account is needed, just use your github account to sign in)
```
[<p align="center"><kbd> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Click here to visit the Notepad++ Community Forum &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; </kbd></p>](https://notepad-plus-plus.org/community/)

**Comment by sasumner** [comment]
The reason this happens is because the Notepad++ does a 2-step procedure:  First it looks through all the tabs for those it needs to save, prompting the user.  Once that part (the saving, or not) is complete for all of the tabs, the second part runs to close the needed files.  One can see this 2-step thing in progress:  Have two or more modified files and invoke the command.  N++ will prompt you to save a file (mentally note which one) and respond to the prompt (with either Yes or No).  N++ will then prompt you for the next file, but at this point you will still see the first file (i.e., it hasn't been closed).  Only when you are finished answering all the prompts will the tabs that you were prompted about be closed/removed.

IF, however, you answer a prompt with "Cancel", the entire process is aborted and everything is left as it currently is at the moment of cancelling.
